File: backend/models/Section.js
const mongoose = require('mongoose');

const sectionSchema = new mongoose.Schema({
  sectionCode: { type: String, required: true, unique: true, uppercase: true },
  name: { type: String, required: true },
  department: { type: String, required: true },
  year: { type: Number, required: true },
  semester: { type: Number, required: true },
  strength: { type: Number, default: 60 },
  academicYear: { type: String },
  classTeacher: { type: String }, // Faculty assigned as class teacher
  isActive: { type: Boolean, default: true }
}, { timestamps: true });

module.exports = mongoose.model('Section', sectionSchema);

File: flutter_app/lib/models/section.dart
class Section {
  final String? id;
  final String sectionCode;
  final String name;
  final String department;
  final int year;
  final int semester;
  final int strength;
  final String? academicYear;
  final bool isActive;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Section({
    this.id,
    required this.sectionCode,
    required this.name,
    required this.department,
    required this.year,
    required this.semester,
    this.strength = 60,
    this.academicYear,
    this.isActive = true,
    this.createdAt,
    this.updatedAt,
  });

  factory Section.fromJson(Map<String, dynamic> json) {
    return Section(
      id: json['_id'],
      sectionCode: json['sectionCode'] ?? '',
      name: json['name'] ?? '',
      department: json['department'] ?? '',
      year: json['year'] ?? 1,
      semester: json['semester'] ?? 1,
      strength: json['strength'] ?? 60,
      academicYear: json['academicYear'],
      isActive: json['isActive'] ?? true,
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
      updatedAt: json['updatedAt'] != null ? DateTime.parse(json['updatedAt']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) '_id': id,
      'sectionCode': sectionCode,
      'name': name,
      'department': department,
      'year': year,
      'semester': semester,
      'strength': strength,
      'academicYear': academicYear,
      'isActive': isActive,
    };
  }
}
